config/gitlab.yml.example

```diff
@@ -89,6 +79,57 @@ production: &base
     # The default is 'tmp/repositories' relative to the root of the Rails app.
     # repository_downloads_path: tmp/repositories
 
+  ## Settings for sending emails
+  outgoing_emails:
+
+    # Set to false if you need to disable email sending from GitLab (default: true).
+    enabled: true
+
+    # Email address and name used in the "From" field in mails sent by GitLab.
+    from: example@example.com
+    display_name: GitLab
+
+    # Email address used in "Reply-To" field in mails sent by GitLab.
+    reply_to: noreply@example.com
+
+    # Defines a delivery method. Possible values are:
+    #   :sendmail      - send emails using sendmail command (default).
+    #   :smtp          - send emails using SMTP server.
+    #   :letter_opener - open sent emails in browser.
+    #   :test          - save emails to ActionMailer::Base.deliveries array.
+    delivery_method: :sendmail
+
+    # Settings for the :sendmail delivery method.
+    sendmail_settings:
+      # The location of the sendmail executable (default: /usr/sbin/sendmail).
+      location: /usr/sbin/sendmail
+
+      # The command line arguments to be passed to sendmail (default: -i -t).
+      arguments: '-i -t'
+
+    # Settings for the :smtp delivery method.
+    smtp_settings:
+      # Hostname of the SMTP server used to send emails (default: localhost).
+      address: localhost
+
+      # Port of the SMTP server used to send emails (default: 25).
+      port: 25
+
+      # If your mail server requires authentication, you need to specify the
+      # authentication type here. Possible values are:
+      #   :plain    - send the password in the clear text.
```

Comment:
I see your point and I agree, but we have two methods that behave almost the same, `:plain` and `:login` and they are both base64 encoded. The way it is written now, one might think `:login` is stronger than `:plain`.

I don't know how you define cleartext, but in my mind it means that a password is read as is. However in base64 you'd have to do a decode first, although trivial.